core/src/program/blocks/mod.rs

```diff
@@ -1,4 +1,5 @@
 use super::{hasher, Digest, Felt, FieldElement, Operation};
+use crate::utils::collections::Vec;
```

Comment:
nit: could we put the `Box` import here as well. This way, in sub-modules (e.g., `join_block.rs`, `loop_block.rs`) we  can just import `Box` from `super`.

Reply:
I now notice that you tend to use the `super::` pattern to propagate imports from the top level.  Would you prefer I refactor the whole PR to align with this pattern?

Reply:
I've refactored where there wasn't too much friction.